### DISCUSSION TITLE
React Native support

### DISCUSSION THREAD

**Discussion by dpsharma1988** [tradingview/lightweight-charts#733] [Offtopic]
I would like to ask you if you are providing any react-native solution for mobile apps.
Please do help if any TradingView library available for react-native which I can use for mobile app development.

**Comment by timocov** [comment]
I think that all of tradingview's charting solution are suitable for react-native. [Here](https://github.com/tradingview/charting-library-examples/tree/master/react-native) you can find an example for charting_library, I think that for lightweight-charts it would be similar or even simpler.

**Reply by dpsharma1988** [reply]
4. Copy all files from https://github.com/tradingview/charting_library/. The earliest supported version of the Charting Library is 17. If you get 404 then you need to request an access to this repository.


This is not working. I will be thankful if you can help out for this.

**Reply by timocov** [reply]
> This is not working

What exactly? What do you get after this step? What version of the library you have after copying?

**Reply by dpsharma1988** [reply]
When I am clicking "request an access to this repository" link, it will open your main website link.
Please let me know what do you mean by "request an access to this repository"?
There is an option of "Get Library" on the website, when I clicked it, again opening Github Code link.

Please answer if your trading view library is available in react-native for mobile apps.
We already have reputed trading apps in the market. We have started a trading application in react-native and we need a react-native chart library for mobile apps.

**Reply by timocov** [reply]
> Please let me know what do you mean by "request an access to this repository"?

If we're talking about charting_library product (not this lightweight-charts library), it's closed-source library and you have to request an access to get it. Don't forget to select this product in the list (see below):

![image](https://user-images.githubusercontent.com/3112183/116874333-14e67080-ac11-11eb-8fca-0611697b9c75.png)

> Please answer if your trading view library is available in react-native for mobile apps.

It can work in react native, but we don't have a react native wrapper for lightweight-charts, only native ones (for ios and android).

**Reply by dpsharma1988** [reply]
Hi,

I have requested for the library access and it says :

*Please note:*

The invite to the dedicated GitHub repository will be sent to your GitHub
account email.

How much time will it take to give the permissions for the library in react-native?

**Reply by dpsharma1988** [reply]
I am waiing for the library access. Please let me know how much time it will take..?

**Reply by urmybestfriend** [reply]
Did you get trading view library for react native?
If you get, please send me please.

**Reply by ahmad7744** [reply]
I also need this library please?

**Reply by salman-ar-sar** [reply]
> I also need this library please?

I have integrated it using WebView using the HTML version.

**Reply by ahmad7744** [reply]
Could you share your repo link with me? or how I can use in my react native project?

**Reply by salman-ar-sar** [reply]
> Could you share your repo link with me? or how I can use in my react native project?

Sorry I can't share it. It is a work project. 

I just created an HTML file with the charting library in script tag and created it there.

**Reply by ahmad7744** [reply]
ok could you share the HTML file with me?

**Reply by salman-ar-sar** [reply]
@ahmad7744 Can't share that as well. I followed this [tutorial](https://tradingview.github.io/lightweight-charts/tutorials/customization/intro).

Here's a sample HTML file from it: https://tradingview.github.io/lightweight-charts/2d783e20a8f4da303bc398d9892da9a6.html

**Comment by codedbystability** [comment]
same situation here . i have signed up but can not find files to copy from the link you provided (https://github.com/tradingview/charting_library/)

**Reply by timocov** [reply]
To be honest I have no idea why you decided to ask your question in this repo since charting library and lightweight-charts are different products

**Reply by Stackerr** [reply]
so, how can i adjust lightweight chart library in react-native? 
install module and use react-native webview?

**Reply by timocov** [reply]
@Stackerr did you try your steps? I'm not familiar with react-native so if you are please give it a try. Also you can look at https://github.com/tradingview/charting-library-examples/tree/master/react-native for example of using charting library with react-native, probably with lightweight-charts it should be simpler.

**Reply by salman-ar-sar** [reply]
@Stackerr Could you get it to work?
@timocov The example isn't working. Is there any guide for integrating lightweight charts on react native?

**Comment by Tsikitsiky** [comment]
Is there anyone who could integrate this charting library with react-native? If there is, could you give a guidance on how to integrate it? 
I have tried the example at https://github.com/tradingview/charting-library-examples/tree/master/react-native but didn't work for me.
Thank you.

**Reply by romfrancois** [reply]
> Is there anyone who could integrate this charting library with react-native? If there is, could you give a guidance on how to integrate it? I have tried the example at https://github.com/tradingview/charting-library-examples/tree/master/react-native but didn't work for me. Thank you.

We've just updated the examples and react-native was one of them.
Could you please give it another try?
Thanks

**Reply by salman-ar-sar** [reply]
@romfrancois Thanks for the update. Tried the new example. Working fine with those instructions. I can view TradingView.

But I need lightweight charts in my use cases. The API in the docs are for HTML elements and Android/iOS native. Is there anyway to use this on React Native?

**Reply by maryjenel** [reply]
Am I the only one getting Page not found for https://github.com/tradingview/charting_library/ ?
cc @romfrancois

**Reply by edew** [reply]
> Am I the only one getting Page not found for https://github.com/tradingview/charting_library/ ?


The page will 404 if you do not have access. You can request access on this page: https://www.tradingview.com/advanced-charts/

**Comment by tomsnep** [comment]
One of the things you could consider trying to integrate this in React Native is to render the library inside a webview (that's basically what the [ios](https://tradingview.github.io/lightweight-charts/docs/ios)/android wrappers do as well). 

Adding local html/css/js to a webview can be a cumbersome DX experience. But i found a way that seems to works really wel (still experimenting though).

What i did:

- Install [`react-native-react-bridge`](https://github.com/inokawa/react-native-react-bridge)
- Implement the lightweight-chart in a React (web) (see example of `react-native-react-bridge`) component (you can just add the package to your RN repo and import it inside the React web component file)
- Interact with the chart from RN context via the message emitter of the `react-native-react-bridge` 

Let me know if you have any questions

**Reply by nanthandev12** [reply]
hello
do you have any example of the implementation

**Comment by KarimScript** [comment]
Hello ! , since you talked about it , can we implement TradingView with react native expo app without using WebView ?

**Reply by sem4phor** [reply]
@romfrancois

**Reply by romfrancois** [reply]
Unfortunately that's not on our roadmap for now as there's only a handful of requests. If you were to successfully achieve that we would be more than happy to inspect the code and credit you with your example online.

**Comment by salman-ar-sar** [comment]
Could anyone run the lightweight charts library on React Native by using just WebView and not React bridge?

**Reply by tomsnep** [reply]
Yes it is possible, i've used [react-native-react-bridge](https://github.com/inokawa/react-native-react-bridge)

**Reply by salman-ar-sar** [reply]
I wanted a way without using React bridge. How is the performance using it?

**Reply by tomsnep** [reply]
Performance and DX is good. It works really well in my opinion. Sending messages from Webview back to RN context is pretty fast

**Reply by salman-ar-sar** [reply]
Ok thanks man!
